src/sentry/migrations/0229_add_frontend_manifest.py

```diff
@@ -0,0 +1,104 @@
+# Generated by Django 2.2.23 on 2021-09-09 16:13
+
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.jsonfield
+
+
+def _make_deferrable(apps, schema_editor):
+    """
+    Change the unique constraint to be deferrable
+    """
+    # Get the db name of the constraint
+    FrontendManifest = apps.get_model("sentry", "FrontendManifest")
+    CONSTRAINT_NAME = schema_editor._constraint_names(
+        FrontendManifest, ["is_production"], unique=True
+    )[0]
+    TABLE_NAME = FrontendManifest._meta.db_table
+
+    # Drop then re-add with deferrable as ALTER doesnt seem to work for unique constraints in psql
+    with schema_editor.connection.create_cursor() as curs:
+        curs.execute(f'ALTER TABLE {TABLE_NAME} DROP CONSTRAINT "{CONSTRAINT_NAME}";')
+        curs.execute(
+            f"ALTER TABLE {TABLE_NAME} ADD CONSTRAINT"
+            f" {CONSTRAINT_NAME}"
+            f" UNIQUE (col1, col2) DEFERRABLE INITIALLY DEFERRED;"
+        )
+
+
+def _unmake_deferrable(apps, schema_editor):
+    """
+    Reverse the unique constraint to be not deferrable
+    """
+    # Get the db name of unique constraint
+    FrontendManifest = apps.get_model("myapp", "FrontendManifest")
+    CONSTRAINT_NAME = schema_editor._constraint_names(
+        FrontendManifest, ["is_production"], unique=True
+    )[0]
+    TABLE_NAME = FrontendManifest._meta.db_table
+
+    with schema_editor.connection.create_cursor() as curs:
+        curs.execute(f'ALTER TABLE {TABLE_NAME} DROP CONSTRAINT "{CONSTRAINT_NAME}";')
+        curs.execute(
+            f"ALTER TABLE {TABLE_NAME} ADD CONSTRAINT"
+            f" {CONSTRAINT_NAME}"
+            f" UNIQUE (col1, col2) NOT DEFERRABLE;"
+        )
```

Comment:
Unfortunately, [deferrable](https://docs.djangoproject.com/en/3.2/ref/models/constraints/#deferrable) was not added until 3.1 -- it might be better to make the partial unique constraint in SQL instead of defining it in the model.